src/coreclr/jit/codegenxarch.cpp

```diff
@@ -7450,28 +7448,11 @@ void CodeGen::genFloatToIntCast(GenTree* treeNode)
     var_types dstType = treeNode->CastToType();
     var_types srcType = op1->TypeGet();
     assert(varTypeIsFloating(srcType) && !varTypeIsFloating(dstType));
+    assert(!varTypeIsSmall(dstType));
 
-    // We should never be seeing dstType whose size is neither sizeof(TYP_INT) nor sizeof(TYP_LONG).
-    // For conversions to byte/sbyte/int16/uint16 from float/double, we would expect the
-    // front-end or lowering phase to have generated two levels of cast. The first one is
-    // for float or double to int32/uint32 and the second one for narrowing int32/uint32 to
-    // the required smaller int type.
-    emitAttr dstSize = EA_ATTR(genTypeSize(dstType));
-    noway_assert((dstSize == EA_ATTR(genTypeSize(TYP_INT))) || (dstSize == EA_ATTR(genTypeSize(TYP_LONG))));
-
-    // We shouldn't be seeing uint64 here as it should have been converted
-    // into a helper call by either front-end or lowering phase, unless we have AVX512
-    // accelerated conversions.
-    assert(!varTypeIsUnsigned(dstType) || (dstSize != EA_ATTR(genTypeSize(TYP_LONG))) ||
-           compiler->canUseEvexEncodingDebugOnly());
-
-    // If the dstType is TYP_UINT, we have 32-bits to encode the
-    // float number. Any of 33rd or above bits can be the sign bit.
-    // To achieve it we pretend as if we are converting it to a long.
-    if (varTypeIsUnsigned(dstType) && (dstSize == EA_ATTR(genTypeSize(TYP_INT))) && !compiler->canUseEvexEncoding())
-    {
-        dstType = TYP_LONG;
-    }
+    // Unless AVX10.2 saturating conversion instructions are available, these
+    // casts should have been lowered to a sequence of HWIntrinsic nodes.
+    assert(compiler->compIsaSupportedDebugOnly(InstructionSet_AVX10v2));
```

Comment:
Do we even "need" this method anymore? Why not just have lowering always introduce the `GenTreeHWIntrinsic` representing the conversion instead?

Reply:
I can look at that in a followup if you don't mind.  I wanted to do one more PR in this area, to move the x64 SSE2 ulong->floating fallback logic from `CodeGen::genIntToFloatCast` to lowering and change it to a branchless implementation.